doc/ecs/components/animation_component.md

```diff
@@ -3,18 +3,53 @@ title: "Animation Component"
 ---
 
 ## Wofür
-- Wofür braucht es diese Comonent?
-- Was macht es? 
+
+Das `AnimationComponent` ist dafür zuständig, die Animationen einer Entität zu verwalten. Es arbeitet mit dem `DrawSystem`
+zusammen, welches im Endeffekt dafür sorgt, dass irgendetwas auf dem Bildschirm angezeigt wird. Es gibt immer eine aktuelle
+Animation, die von einer Entität abgespielt wird. Diese Animation kann während der Laufzeit durch die entsprechenden Methodenaufrufe
+geändert werden.
 
 ## Aufbau
 
-- UML
-- Aufbau Erklären
-- Erklären warum Animationen auch in anderen Components gespeichert werden
+![UML Diagram](img/animationComponent.png)
 
 ## Wie nutzt man es
-- Welche Prameter gibt es, was machen diese? 
 
-## Testabdeckung
-- Wie ist die Testabdeckung?
-- Welche Methode/Funktionalitäten können nicht geteste werden, warum? 
\ No newline at end of file
+### Erzeugen - Möglichkeit 1
+
+Erzeugen eines neuen `AnimationComponent`-Objektes mithilfe des einfachen Konstruktors `AnimationComponent(entity: Entity)`. Hier werden alle Animationen
+auf die Standardwerte, daher auf eine "missing texture" Animation gesetzt.
+
+### Erzeugen - Möglichkeit 2
+
+Erzeugen eines neuen `AnimationComponent`-Objektes mithilfe des komplexeren Konstruktors `AnimationComponent(entity: Entity, idleAnimation: Animation)`. Die hier
+übergebene Idle-Animation wird für beide Richtungen, daher für `idleLeft` und `idleRight` verwendet.
+
+### Erzeugen - Möglichkeit 3
+
+Erzeugen eines neuen `AnimationComponent`-Objektes mithilfe des komplexeren Konstruktors `AnimationComponent(entity: Entity, idleLeft: Animation, idleRight: Animation)`.
+Hier werden die Animationen entsprechend der Parameternamen gesetzt.
+
+## Methoden
+
+### setCurrentAnimation(animation: Animation): void
+
+Setzt die aktuelle Animation der Entität. Die hier übergebene Animation muss ***NICHT*** aus dem `AnimationComponent` stammen.
+
+#### Parameter
+
+| Parametername | Typ       | Beschreibung                                 |
+|---------------|-----------|----------------------------------------------|
+| animation     | Animation | Die Animation, welche angezeigt werden soll. |
```

Comment:
Das würde ich als Text schreiben (ein kurzer Satz) und auf die Überschrift und die Tabelle verzichten.

Reply:
Edit: Andererseits machst Du das überall so, sehe ich grad. 

@AMatutat War das die Vorgabe, also kommt das in den anderen Docs auch so? Dann sollten wir es so lassen. Anderenfalls würde ich diese Überschriften, die die einzigen ihrer Ebene sind (hier wäre das die einzige H4-Überschrift im aktuellen Abschnitt), rausnehmen und den Text darunter integrieren - sowas wird normalerweise als Gliederungsfehler betrachtet.

Reply:
nach unseren neuen Konzept fliegt das doch vermutlich eh alles in die javadoc, oder? 
Würde das dann jetzt erstmal so lassen und ich übertrag das beim Überarbeiten.

Reply:
Ja, gut. Dann mache ich diesen Kommentar schnell wieder zu :)